cutlery/_wordpiece.pyx

```diff
@@ -37,18 +42,112 @@ cdef class WordPieceProcessor:
             token_ids.append(match.piece_id)
             token_pieces.append(f"{prefix}{token[:match.prefix_len]}")
             token = token[match.prefix_len:]
-            pieces = &self.continuation_pieces
+            pieces = &self.continuation_piece_to_id
             prefix = "##"
         return token_ids, token_pieces
 
+    def decode(self, pieces: Iterable[int], *, unk_token: str = "<unk>") -> str:
+        """
+        Decode piece identifiers into string. Invalid piece identifiers
+        are replaced with the `unk_token` string.
+
+            ids (Iterable[int]): Piece IDs.
+            unk_token (str): Marker for unknown piece IDs.
+            RETURNS (str): Decoded string.
+        """
+        token_pieces = [self.piece_id_to_str(id, unk_token=unk_token) for id in pieces]
+        return "".join(token_pieces)
+
     def get_initial(self, piece: str) -> int:
```

Comment:
I think there is a lot of overlap between the next few methods. It may be nicer to have a single method `piece_id_to_str` that returns a tuple with the piece and whether it is initial.